airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/DefaultTyperDeduper.kt

```diff
@@ -377,7 +377,7 @@ class DefaultTyperDeduper<DestinationState : MinimumDestinationState>(
     }
 
     @Throws(Exception::class)
-    override fun typeAndDedupe(streamSyncSummaries: Map<StreamDescriptor, StreamSyncSummary>) {
+    override fun<T> typeAndDedupe(streamSyncSummaries: Map<StreamDescriptor, T>) where T: StreamSyncSummary{
```

Comment:
@colesnodgrass I was having an issue where the java calling this was using a `Map<StringDescriptor, ? extends StreamSyncSummary>`.
This works, but I'm not sure if there's a cleaner way to do the same thing?

Reply:
This doesn't seem correct, it should use the `? extends` only if this was defined as `Map<StringDescriptor, out StreamSyncSummary>`.  Unless I'm misunderstanding what you're saying, and you actually want to add the `out` covariant keyword.

As per the workaround (and if [variance](https://kotlinlang.org/docs/generics.html) doesn't need to be defined) you should be able to clean it up a bit by
```kotlin
override fun<T : StreamSyncSummary> typesAndDedupe(streamSyncSummaries: Map<StreamDescriptor, T>)
```